### PR TITLE
feat: adds visibility option

### PR DESCRIPTION
**Pull request by rgomeztao**
[TR-1328](https://oat-sa.atlassian.net/browse/TR-1328)

changelog:
* disable visibility of selection property in section and sub section
* creates a new section property to allow visibility via migration

Steps to check:

* login with admin credentials
* go to tests section
* create a test
* add some items
* check the selection tool
* To enable the “with replacement“ (out of the scope, but for the sake of tests), editing the config/tao/client_lib_config_registry and adding 'services/features' => [    'visibility' => [        'taoQtiTest/creator/properties/selectionWithReplacement' => 'show',    ]] just at the end, and hard reloading your browser should show the “with replacement“ option back

[TR-1328]: https://oat-sa.atlassian.net/browse/TR-1328?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: views/js/controller/creator/templates/section-props.tpl [hunk 1]
```diff
@@ -153,7 +153,7 @@
         <div class="category-presets"></div>
     </div>
 
-    <h4 class="toggler closed" data-toggle="~ .section-selection">{{__ 'Selection'}}</h4>
+    <h4 class="toggler closed" data-toggle="~ .section-selection">{{__ 'Selection'}} </h4>
```
Comment (gabrielfs7) [inline views/js/controller/creator/templates/section-props.tpl:156]: I suggest to revert this change, if is just a space :) 

```suggestion
    <h4 class="toggler closed" data-toggle="~ .section-selection">{{__ 'Selection'}}</h4>
```

File: views/js/controller/creator/views/subsection.js [hunk 2]
```diff
@@ -81,6 +83,11 @@ define([
         if (!_.isEmpty(config.routes.blueprintsById)) {
             subsectionModel.hasBlueprint = true;
         }
+
+        if (servicesFeatures.isVisible('taoQtiTest/creator/properties/selectionWithReplacement', false)) {
+            sectionModel.hasSelectionWithReplacement = true;
```
Comment (gabrielfs7) [inline views/js/controller/creator/views/subsection.js:88]: Do we need to initialise `hasSelectionWithReplacement` to avoid undefined cases?

File: views/js/controller/creator/templates/section-props.tpl (hunk 1 above)
Comment (rgomeztao) [reply views/js/controller/creator/templates/section-props.tpl]: fixed!! thanks @gabrielfs7   !

File: views/js/controller/creator/views/subsection.js (hunk 2 above)
Comment (rgomeztao) [reply views/js/controller/creator/views/subsection.js:88]: thats correct, but according to the template engine (handlebars) thats just fine. This behaviour was confirmed on other cases of the same view 
https://handlebars-archive.knappi.org/builtin_helpers.html

"You can use the if helper to conditionally render a block. If its argument returns false, undefined, null, "", 0, or [], Handlebars will not render the block."

**Review by hectoras — COMMENTED** [review]
Please change the target branch to `develop` :slightly_smiling_face: 

- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Target branch is correct
- [x] Commits are following conventional commits
- [ ] New code is covered by tests (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [ ] Kitchen links are not exposed in the description
- [x] No unnecessary dependencies are installed

**Comment by rgomeztao** [comment]
> Please change the target branch to `develop` slightly_smiling_face
> 
> * [x]  Pull request title and description are meaningful
> * [x]  Pull request's target is not `master`
> * [x]  Target branch is correct
> * [x]  Commits are following conventional commits
> * [ ]  New code is covered by tests (if applicable)
> * [x]  New code is respecting code style rules
> * [x]  New code is respecting best practices
> * [x]  New code is not subject to concurrency issues (if applicable)
> * [ ]  Feature is working correctly on my local machine (if applicable)
> * [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
> * [x]  Acceptance criteria are respected
> * [ ]  Kitchen links are not exposed in the description
> * [x]  No unnecessary dependencies are installed

updated ;), thanks!

**Review by hectoras — APPROVED** [review]

**Review by jsconan — APPROVED** [review]
It looks ok.
However, I have a few remarks:
- there is no migration script for setting the visibility registry, is it ok?
- the PR description does not detail what to configure for testing the feature...
    something like:
    - open the file `config/tao/client_lib_config_registry.conf.php`
    - add:
        ```
        'services/features' => [
            'visibility' => [
                'taoQtiTest/creator/properties/selectionWithReplacement' => 'show' // or 'hide'
            ]
        ]
        ```
- please avoid using a `push-force` to update a PR: it masquerades the history, it may impact other developers breaking their local clone, etc.
- conceptually, the code looks ok, but I'd instead use a direct assignment from the API rather than an orphan IF than may leave an inconsistent dataset (the property is set or undefined). Ex: `props.name = registry.isVisible(...)` instead of `if (registry.isVisible(...)) props.name = true`

**Review Checklist**
- [ ] ~~New code is covered by tests (if applicable)~~ _- not applicable since the change applies to a controller._
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] ~~New code is not subject to concurrency issues (if applicable)~~
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Commits messages are meaningful
- [x] Commits are atomic

**Comment by rgomeztao** [comment]
@js to answer your questions :
- no migrations to allow certain customers to see this option, as it is buggy and never worked as expected :)
- the information you miss is not here, but available in a ticket comment i left a few minutes ago :), ill copy it here as well, thanks!
- normally i dont just ---force, but -force-with-lease. On any case im more a follower of atomic commits, but had the feeling that it was not the case here. ill not force next time :), thanks!
- finally about the if sentence, im going to implement your change, as it is an improvement in cyclomatic complexity. ill push a change as soon as i test both kitchen and local env :), thanks!

File: views/js/controller/creator/views/subsection.js (hunk 2 above)
Comment (rgomeztao) [reply views/js/controller/creator/views/subsection.js]: updated to fill the value and reduce the amount of if sentences :), thanks @gabrielfs7 !
